File: backend/internal/models/albums.go
package models

type AlbumImagesAssignments map[int]map[string]int // album_id -> images -> rank

type AlbumPatchRequest struct {
	Rank *int `json:"rank"`
}

type Album struct {
	ID             int     `json:"id"`
	Name           string  `json:"name"`
	CoverImageUUID *string `json:"cover_image_uuid"`
	Rank           int     `json:"rank"`
}

File: frontend/client/src/types/albums.ts
interface Album {
    id: number;
    name: string;
    cover_image_uuid: string | null;
    rank: number;
}

interface FetchedAlbumImageAssignment {
    [key: string]: {
        [key: string]: number;
    }
}

interface AlbumImageAssignment {
    [key: number]: {
        [key: string]: number;
    }
}

export { Album, FetchedAlbumImageAssignment, AlbumImageAssignment }